include/swift-c/DependencyScan/DependencyScan.h

```diff
@@ -0,0 +1,228 @@
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_C_DEPENDENCY_SCAN_H
+#define SWIFT_C_DEPENDENCY_SCAN_H
+
+#include "DSString.h"
+#include "DependencyScanMacros.h"
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/// The version constants for the SwiftDependencyScan C API.
+/// DEPSCAN_VERSION_MINOR should increase when there are API additions.
+/// DEPSCAN_VERSION_MAJOR is intended for "major" source/ABI breaking changes.
+#define DEPSCAN_VERSION_MAJOR 0
+#define DEPSCAN_VERSION_MINOR 1
+
+DEPSCAN_BEGIN_DECLS
+
+//=== Public Scanner Data Types -------------------------------------------===//
+
+typedef enum {
+  DEPSCAN_DEPENDENCY_INFO_SWIFT_TEXTUAL = 0,
+  DEPSCAN_DEPENDENCY_INFO_SWIFT_BINARY = 1,
+  DEPSCAN_DEPENDENCY_INFO_SWIFT_PLACEHOLDER = 2,
+  DEPSCAN_DEPENDENCY_INFO_CLANG = 3
+} ds_dependency_info_kind_t;
+
+/// Opaque container of the details specific to a given module dependency.
+typedef void *ds_module_details_t;
+
+typedef struct {
+  /// The module's name
+  /// The format is:
+  /// `<module-kind>:<module-name>`
+  /// where `module-kind` is one of:
+  /// "swiftTextual"
+  /// "swiftBinary"
+  /// "swiftPlaceholder"
+  /// "clang""
+  ds_string_t module_name;
+
+  /// The path for the module.
+  ds_string_t module_path;
+
+  /// The source files used to build this module.
+  ds_string_set_t *source_files;
+
+  /**
+   * The list of modules which this module direct depends on.
+   * The format is:
+   * `<module-kind>:<module-name>`
+   */
+  ds_string_set_t *direct_dependencies;
+
+  /// Specific details of a particular kind of module.
+  ds_module_details_t details;
+} ds_dependency_info_t;
+
+/// Full Dependency Graph (Result)
+
+typedef struct {
+  int count;
+  ds_dependency_info_t *modules;
+} ds_dependency_set_t;
+
+typedef struct {
+  /// The name of the main module for this dependency graph (root node)
+  ds_string_t main_module_name;
+
+  /// The complete list of modules discovered
+  ds_dependency_set_t *module_set;
+} ds_dependency_result_t;
+
+typedef struct {
+  /// The complete list of imports discovered
+  ds_string_set_t *import_set;
+} ds_prescan_result_t;
+
+//=== Batch Scan Input Specification --------------------------------------===//
+
+typedef struct {
+  ds_string_t module_name;
+  ds_string_t arguments;
+  bool is_swift;
+} ds_batch_scan_entry_t;
+
+typedef struct {
+  int count;
+  ds_batch_scan_entry_t *modules;
+} ds_batch_scan_input_t;
+
+typedef struct {
+  int count;
+  ds_dependency_result_t **results;
+} ds_batch_scan_result_t;
+
+//=== Dependency Result Functions -----------------------------------------===//
+
+DEPSCAN_PUBLIC ds_dependency_info_kind_t
+ds_get_module_detail_kind(ds_module_details_t details);
+
+//=== Swift Textual Module Details query APIs -----------------------------===//
+DEPSCAN_PUBLIC ds_string_t
+ds_get_swift_textual_detail_module_interface_path(ds_module_details_t details);
```

Comment:
I'd suggest to try to stick to this naming convention for the functions: \<prefix>\_\<object>\_\<action>
So in this case: `ds_swift_textual_detail_get_module_interface_path`

Reply:
Made all API consistently follow this naming scheme, thank you.